src/reducers/controlFieldUtils.js

```diff
@@ -93,3 +95,51 @@ export function fillControlFieldGaps(baseField, sourceField, min = 0, max = 39)
 
   baseField.value = mergedCharArray.join(''); // eslint-disable-line functional/immutable-data
 }
+
+export function genericControlFieldCharPosFix(baseField, sourceField, typeOfMaterial, supportedTypesOfMaterial, legalValues, position, valueForUnknown, noAttemptToCode) { // eslint-disable-line max-params
+  // Initially written fro field 008, but may be applied to 006 and 007 as well (I guess).
+  if (supportedTypesOfMaterial.length > 0 && !supportedTypesOfMaterial.includes(typeOfMaterial)) {
```

Comment:
Check typeOfMaterial for both base and source here?

Reply:
(no longer needed because of the change described in the next comment.)

Reply:
Both base and source are now checked. Great call!

(Also supportedTypesOfMaterial.length > 0 requirement is now longer used. We assume it to be always > 0, and 0 will fail on purpose.)